atlasdb-cassandra-multinode-tests/src/test/java/com/palantir/cassandra/multinode/NodesDownTestSetup.java

```diff
@@ -15,46 +15,62 @@
  */
 package com.palantir.cassandra.multinode;
 
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace.GLOBAL;
+
 import com.google.common.base.Throwables;
 import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
 import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceRuntimeConfig;
 import com.palantir.atlasdb.cassandra.ImmutableCassandraKeyValueServiceConfig;
-import com.palantir.atlasdb.containers.Containers;
+import com.palantir.atlasdb.containers.ContainersV2;
 import com.palantir.atlasdb.containers.ThreeNodeCassandraCluster;
 import com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueService;
 import com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueServiceImpl;
 import com.palantir.docker.compose.connection.DockerPort;
 import com.palantir.refreshable.Refreshable;
 import java.io.IOException;
 import java.time.Duration;
-import java.util.List;
+import java.util.Set;
 import org.awaitility.Awaitility;
-import org.junit.AfterClass;
-import org.junit.ClassRule;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.reflections.Reflections;
 
-public abstract class NodesDownTestSetup {
+public class NodesDownTestSetup implements BeforeAllCallback, ExtensionContext.Store.CloseableResource {
+    private static volatile boolean started = false;
     private static final int CASSANDRA_THRIFT_PORT = 9160;
     private static final CassandraKeyValueServiceConfig CONFIG = ImmutableCassandraKeyValueServiceConfig.copyOf(
                     ThreeNodeCassandraCluster.KVS_CONFIG)
             .withSchemaMutationTimeoutMillis(3_000);
     private static final Refreshable<CassandraKeyValueServiceRuntimeConfig> RUNTIME_CONFIG =
             ThreeNodeCassandraCluster.KVS_RUNTIME_CONFIG;
 
-    @ClassRule
-    public static final Containers CONTAINERS =
-            new Containers(NodesDownTestSetup.class).with(new ThreeNodeCassandraCluster());
+    public static final ContainersV2 CONTAINERS =
+            new ContainersV2(NodesDownTestSetup.class).with(new ThreeNodeCassandraCluster());
+
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception {
+        synchronized (NodesDownTestSetup.class) {
```

Comment:
can't you just make this method `public void synchronized beforeAll(...` instead of having it in the body?